src/Listing/Bid/Bid.model.js

```diff
@@ -81,12 +106,77 @@ export class Bid extends Model {
     // Re-validate bids for assets with its fingerprint back to the original value
     await this.db.query(
       SQL`UPDATE ${raw(this.tableName)}
-        SET status = ${LISTING_STATUS.open}
+        SET status = ${
+          LISTING_STATUS.open
+        }, block_time_updated_at = ${blockTime}, block_number = ${blockNumber}
         WHERE block_time_created_at <= ${blockTime}
           AND token_address = ${tokenAddress}
           AND token_id = ${tokenId}
           AND status = ${LISTING_STATUS.fingerprintChanged}
           AND fingerprint = ${fingerprint}`
     )
   }
+
+  static async updateAssetOwner(
+    tokenAddress,
+    assetId,
+    seller,
+    blockTime,
+    blockNumber
+  ) {
+    return this.db.query(
+      SQL`UPDATE ${raw(this.tableName)}
+        SET seller = ${seller}, block_time_updated_at = ${blockTime}, block_number = ${blockNumber}
+        WHERE block_time_created_at <= ${blockTime}
+          AND token_address = ${tokenAddress} 
+          AND asset_id = ${assetId}
+          AND status = ANY(${[
+            LISTING_STATUS.open,
+            LISTING_STATUS.fingerprintChanged
+          ]})`
+    )
+  }
+
+  static async findBiddedAssetsByStatus(filters) {
+    const { status, bidder /*, seller  */ } = filters
+    const Assets = new Listing.getListableAssets()
+
+    // @nacho TODO: temp code. move to estate.queries or asset
+    // prettier-ignore
+    const estateHasParcels = () => 
```

Comment:
Is this a function because you plan to extract it? if so it's ok, but it could be a const otherwise